test/core/test_lwt_mvar.ml

```diff
@@ -0,0 +1,73 @@
+(* OCaml promise library
+ * http://www.ocsigen.org/lwt
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, with linking exceptions;
+ * either version 2.1 of the License, or (at your option) any later
+ * version. See COPYING file for details.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *)
+
+open Test
+open Lwt
+
+let suite =
+  suite "lwt_mvar" [
+    test "basic take"
+      (fun () ->
+         let x = Lwt_mvar.create 0 in
+         let y = Lwt_mvar.take x in
+         return (y = Lwt.return 0)
+      );
+
+    test "blocking put"
+      (fun () ->
+         let x = Lwt_mvar.create 0 in
+         let y = Lwt_mvar.put x 1 >>= (fun _ -> return 1) in
+         return (y <> Lwt.return 1)
```

Comment:
This is optional (the test is correct regardless), but I believe you could do:

```ocaml
(* ... *)
let y = Lwt_mvar.put x 1 in
Lwt.return (Lwt.state y = Lwt.Sleep)
```

This makes the test slightly clearer, and it would be directly testing that `y` is pending. Actually, I would recommend always checking the state of promises with `Lwt.state` rather than comparing to another promise, as comparing with another promise will fail if the internal definition of promises ever changes to include, say, unique ids for tracing.

Sorry about the name `Lwt.Sleep` for "pending," that's a historical holdover :)